### ISSUE TITLE
jsx-space-before-closing should also handle non self closing tags

### ISSUE THREAD

**Issue by k15a** [yannickcr/eslint-plugin-react#920]
Currently `jsx-space-before-closing` only supports self closing tags but I think it should also handle non self closing tags the same way.

The rule should either accept a string with `always` or `never` which enforces the style on booth self and not self closing tags or an object where you can specify a different behaviour for booth.

**Comment by ljharb** [comment]
@k15a can you give an example of some code that you'd want to be warned on?

**Comment by ljharb** [comment]
It seems reasonable for the rule to add an object config format, like this:
```js
["error", { selfClosing: "always" | "never", somethingElse: "ignore" | "always" | "never" }
```

but it's hard to know what "somethingElse" should be, without seeing code :-)
